tools/infrastructure/run_tests.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# Copyright (c) 2019, Ford Motor Company
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following
+# disclaimer in the documentation and/or other materials provided with the
+# distribution.
+#
+# Neither the name of the Ford Motor Company nor the names of its contributors
+# may be used to endorse or promote products derived from this software
+# without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+run_test() {
+    full_test_name="$1"
+    cur_test="$2"
+    test_count="$3"
+    testname=${1##*/}
+    fullpathname=${full_test_name%/*}
+    cur_dir=$PWD
+    echo "      Start $cur_test: $testname"
+
+    cd $fullpathname
+    test_output=`./$testname 2>&1`
+    test_result=`echo $?`
+    cd $cur_dir
+
+    testname=`echo $testname | sed -r ':l;/.{40}/q;s/.*/&./;bl'`
+    cur_test_test_count=`echo $cur_test/$test_count | sed -r ':l;/.{5}/q;s/.*/& /;bl'`
```

Comment:
@BSolonenko I believe, this could be used in **run_all_tests()**, since, as for me, **cur_test** and **test_count** should not be used in a function supposed to run a single test.